src/Autofac/Core/Resolving/Middleware/DefaultMiddlewareConfiguration.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Autofac.Core.Resolving.Middleware;
+
+/// <summary>
+/// Allows configuration of default middleware.
+/// </summary>
+public static class DefaultMiddlewareConfiguration
+{
+    /// <summary>
+    /// Universally disables the circular dependency check on every resolution. Instead,
+    /// a circular dependency may be caught after many loops and may cause undesirable
+    /// side effects, however resolution performance should improve with deep resolution chains.
```

Comment:
Can we add a comment here (perhaps in a `<remarks>` block) to the effect that this should only be done after extensive testing of the application to ensure no circular dependencies exist (or words to that effect).